Branch-SDK/src/main/java/io/branch/referral/util/BranchCPID.java

```diff
@@ -0,0 +1,101 @@
+package io.branch.referral.util;
+
+import android.text.TextUtils;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class BranchCPID {
+
+    JSONObject cpidData;
+
+    public BranchCPID() {
+    }
+
+    public BranchCPID(JSONObject cpidData) {
+        this.cpidData = cpidData;
+    }
+
+    public String getCrossPlatformID() {
+        if (cpidData != null && cpidData.length() != 0) {
+            try {
+                return cpidData.getJSONObject("user_data").getString("cross_platform_id");
+            } catch (JSONException e) {
+                return null;
+            }
+        }
+        return null;
+    }
+
+    public JSONArray getPastCrossPlatformIds() {
+        if (cpidData != null && cpidData.length() != 0) {
+            try {
+                return cpidData.getJSONObject("user_data")
+                        .getJSONArray("past_cross_platform_ids");
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+
+        }
+        return null;
+    }
+
+    public JSONArray getProbabilisticCrossPlatformIds() {
+        if (cpidData != null && cpidData.length() != 0) {
+            try {
+                JSONArray probCPIDs = cpidData.getJSONObject("user_data")
```

Comment:
Lots of Strings that might be better used as constants.  `user_data` is referenced 4 times for instance.